src/app/core/http-services/item-navigation.service.ts

```diff
@@ -149,52 +68,36 @@ const itemNavigationDataDecoder = D.struct({
     title: D.nullable(D.string),
   }),
   type: D.literal('Chapter','Task','Course','Skill'),
-  children: D.array(
-    pipe(
-      D.struct({
-        bestScore: D.number,
-        entryParticipantType: D.literal('User', 'Team'),
-        hasVisibleChildren: D.boolean,
-        id: D.string,
-        noScore: D.boolean,
-        permissions: permissionsDecoder,
-        requiresExplicitEntry: D.boolean,
-        results: D.array(D.struct({
-          attemptAllowsSubmissionsUntil: dateDecoder,
-          attemptId: D.string,
-          endedAt: D.nullable(dateDecoder),
-          latestActivityAt: dateDecoder,
-          scoreComputed: D.number,
-          startedAt: D.nullable(dateDecoder),
-          validated: D.boolean,
-        })),
-        string: D.struct({
-          languageTag: D.string,
-          title: D.nullable(D.string),
-        }),
-        type: D.literal('Chapter','Task','Course','Skill'),
-      }),
-      D.intersect(
-        D.partial({
-          watchedGroup: pipe(
-            D.struct({
-              canView: D.literal('none','info','content','content_with_descendants','solution'),
-            }),
-            D.intersect(
-              D.partial({
-                allValidated: D.boolean,
-                avgScore: D.number,
-              })
-            )
-          ),
-        })
-      )
-    )
-  ),
+  children: D.array(itemNavigationChildDecoder),
 });
 
 export type ItemNavigationData = D.TypeOf<typeof itemNavigationDataDecoder>;
 
+const rootActivitiesDecoder = D.array(
+  D.struct({
+    groupId: D.string,
+    name: D.string,
+    type: D.literal('Class', 'Team', 'Club', 'Friends', 'Other', 'User', 'Session', 'Base', 'ContestParticipants'),
+    activity: itemNavigationChildDecoderBase
+  })
+);
+
+export type RootActivities = D.TypeOf<typeof rootActivitiesDecoder>;
+
+const rootSkillsDecoder = D.array(
+  D.struct({
+    groupId: D.string,
+    name: D.string,
+    type: D.literal('Class', 'Team', 'Club', 'Friends', 'Other', 'User', 'Session', 'Base', 'ContestParticipants'),
+    skill: itemNavigationChildDecoderBase
+  })
+);
+
+export type RootSkills = D.TypeOf<typeof rootSkillsDecoder>;
```

Comment:
Suggestion:
I find it usually easier to declare the non-array type/decoder and then construct the array type or decoder where needed.

It would give 
```ts
const rootSkillDecoder = D.struct(…)
export type RootSkill = D.TypeOf<typeof rootSkillDecoder>
```
and where you need it as array, use `D.array(rootSkillDecoder)` or `RootSkill[]`

Feels cleaner to me, your call

Reply:
I have difficulties to find an actual reason for this. In what case would it be "easier"? (especially as you can always do `MyList[number]` to get the inner type.

Ok, I'll do this, even if not very convinced :-)